pkg/controller/nodes/task/handler.go

```diff
@@ -479,7 +483,21 @@ func (t Handler) invokePlugin(ctx context.Context, p pluginCore.Plugin, tCtx *ta
 		if ee != nil {
 			pluginTrns.ObservedExecutionError(ee)
 		} else {
-			pluginTrns.ObserveSuccess(tCtx.ow.GetOutputPath(), tCtx.ow.GetDeckPath(), &event.TaskNodeMetadata{CacheStatus: cacheStatus.GetCacheStatus(), CatalogKey: cacheStatus.GetMetadata()})
+			var deckUri *storage.DataReference
+			exists, err := tCtx.ow.GetReader().DeckExists(ctx)
+			if err != nil {
+				logger.Errorf(ctx, "Failed to check deck file existence. Error: %v", err)
+				return pluginTrns, regErrors.Wrapf(err, "failed to check existence of deck file")
```

Comment:
I think we don't need to return an error here because the deck file can be non-existent if people use an older version of flytekit or [disable flyte deck](https://github.com/flyteorg/flytekit/blob/27ac89732364fd96d767584fbda1af7696a41ac5/flytekit/core/task.py#L92) in flytekit.

Reply:
DeckExists should not return an error if the file doesn't exist. It should return a nil err and a false value for `exists` output... and that shouldn't be an error I agree.

An error here means we failed to even check the existence of the file, as in "s3 HEAD api call failed"... maybe an auth issue may be a rate limit issue... etc.

Reply:
I see, thanks